torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -827,14 +864,7 @@ class ShapePropagator {
           if (auto maybe_tensor_types =
                   gatherTensorTypes<DimensionedTensorType>(node)) {
             AT_ASSERT(maybe_tensor_types->size() >= 2);
-            auto first_scalar_type = (*maybe_tensor_types)[0]->scalarType();
-            auto second_scalar_type = (*maybe_tensor_types)[1]->scalarType();
-            size_t arg_for_type = 0;
-            if (c10::promoteTypes(first_scalar_type, second_scalar_type) !=
```

Comment:
We could really use more general jit tests, but as I commented elsewhere I wasn't able to generate any valid types with the existing code (regardless of my particular changes to add/div/sub). I added print statements and the existing broadcast never found a type. From my other review comment: "Typically the type of the arg.type() was TensorType or NumberType, not float/int/bool, and so couldn't be used."

Reply:
I agree we could use more `jit` tests. However the lack of tests is IMO a reason to try to make minimal changes here

Reply:
@eelison I separated out the impacted ops. I still think this needs to be updated at some point because I think we don't actually get valid scalar types back in these ops, but that's a separate issue so will retain existing behavior for this PR